src/lib/rules/no-vulnerable-libraries/no-vulnerable-libraries.ts

```diff
@@ -0,0 +1,203 @@
+                }
+            } catch (e) {
+                // There was a problem loading the file --> download
+                needsUpdate = true;
+            }
+
+            if (needsUpdate) {
+                try {
+                    debug('Updating snykDB');
+                    const res = await promisify(request)({ url: 'https://snyk.io/partners/api/v2/vulndb/clientside.json' });
+
+                    await promisify(writeFile)(snykDBPath, res.body, 'utf-8');
+                } catch (e) {
+                    // Problem downloading the remote resource --> use local copy
+                }
+            }
+
+            return JSON.parse(await readFileAsync(snykDBPath));
+        };
+
+        /** If a used library has vulnerability that meets the minimum threshold, it gets reported.  */
+        const reportLibrary = async (library: Library, vulns: Array<Vulnerability>, resource: string) => {
+            let vulnerabilities = vulns;
+
+            if (minimumSeverity !== 'low') {
+                debug('Filtering vulnerabilities');
+                vulnerabilities = vulnerabilities.filter((vulnerability) => {
+                    const { severity } = vulnerability;
+                    let passes = false;
+
+                    switch (minimumSeverity) {
+                        case 'medium':
+                            passes = severity === 'medium' || severity === 'high';
+                            break;
+                        case 'high':
+                            passes = severity === 'high';
+                            break;
+                        default: break;
+                    }
+
+                    return passes;
+                });
+            }
+
+            if (vulnerabilities.length === 0) {
+                return;
+            }
+
+            const vulnerabilitiesBySeverity = groupBy(vulnerabilities, 'severity');
+            const link = `https://snyk.io/vuln/${vulnerabilities[0].packageManager}:${vulnerabilities[0].packageName}`;
+            const detail = Object.entries(vulnerabilitiesBySeverity).map(([severity, entries]) => {
+                return `${entries.length} ${severity}`;
+            })
+                .join(', ');
+
+            if (detail) {
+                await context.report(resource, null, `${library.name}@${library.version} has ${vulnerabilities.length} known vulnerabilities (${detail}). See ${link} for more information.`);
+            }
+        };
+
+        /** Given a list of libraries, reports the ones that have known vulnerabilities. */
+        const detectAndReportVulnerableLibraries = async (libraries: Array<Library>, resource: string) => {
+            // TODO: Check if snykDB is older than 24h and download if so. If not, or if itfails, use local version
+            const snykDB = await loadSnyk();
+
+            for (const lib of libraries) {
+                const snykInfo = snykDB.npm[lib.npmPkgName] as Vulnerability[];
```

Comment:
`Array<Vulnerability>`

Reply:
We have a rule configured to for `as Something` over the other way of casting. Or are you talking about adding the type to `snykInfo`? In that case it is assigned directly because of the `as Vulnerability[]`.

Reply:
> We have a rule configured to for as Something over the other way of casting

That's because of [this](https://github.com/sonarwhal/sonar/blob/bbf1e6eaa401a14733a623a9416292966d8e64e0/.eslintrc.json#L259) (see also: [`no-angle-bracket-type-assertion`](https://github.com/nzakas/eslint-plugin-typescript/blob/f860b4bf3b43e6129fc5ca28564b8ef75a79df5c/docs/rules/no-angle-bracket-type-assertion.md)).

Reply:
I was talking about to change `Vulnerbility[]` to `Array<Vulnerability>` because we are using that way in all the code.
The rule `no-angle-bracket-type-assertion` should be used if our code include `tsx` files [see here](https://github.com/nzakas/eslint-plugin-typescript/blob/master/docs/rules/no-angle-bracket-type-assertion.md) if not, we can remove that rule